tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,76 @@
+import json
+import subprocess
+import urllib.request
+from time import sleep
+
+from tenacity import retry, stop_after_delay, wait_fixed
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(15))
+def test_dcgm_exporter():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "http://localhost:9400/metrics"
+    dcgm_exporter_service = "dcgm.dcgm-exporter"
+
+    subprocess.run(f"sudo snap start {dcgm_exporter_service}".split(), check=True)
```

Comment:
Suggestions to simplify:

- check_call is equivalent to check=True (can make it simpler; optional)
- prefer explicit list rather than building a string then splitting it

```suggestion
    subprocess.check_call(["sudo", "snap", "start", dcgm_exporter_service])
```

Reply:
As I said in another comment there is the [snap-api](https://snapcraft.io/docs/snapd-api#heading--snaps) and I think it's preferable to use it instead of subprocess

Reply:
I will also resolve this. I saw the string splitting in several projects and prefer the same way to make it easier to read the intended command. IMO

Reply:
OK this isn't a big deal, and it's only test code. :+1: